### PR TITLE
Add support to report results to Gitea review comments

### LINKED ISSUE #1295: Add Support of Gitea as Reporter
https://gitea.io/en-us/

was also mentioned in #29

### PR DESCRIPTION
**Pull request by lafriks**
- [x] Updated Unreleased section in [CHANGELOG](https://github.com/reviewdog/reviewdog/blob/master/CHANGELOG.md) or it's not notable changes.

Fixes #1295

attels

Should probably wait for [gitea/go-sdk#638](https://gitea.com/gitea/go-sdk/pulls/638)

### CODE REVIEW

**Review by haya14busa — CHANGES_REQUESTED** [review]

File: service/gitea/gitea.go [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+		}
+
+		body := commentutil.MarkdownComment(c)
+		if g.postedcs.IsPosted(c, giteaCommentLine(c), body) {
+			// it's already posted. skip it.
+			continue
+		}
+
+		if !c.Result.InDiffContext {
+			// If the result is outside of diff context, fallback to Gitea Review
```
[the remark this answers is not in the document]
Comment (lafriks) [reply service/gitea/gitea.go]: @haya14busa I removed fallback.

Yes, I did look on logic in GitHub as Gitea API is designed similar to GitHubs.

File: README.md [hunk 2]
```diff
@@ -931,6 +932,7 @@ so reviewdog will use [Check annotation](https://docs.github.com/en/rest/checks/
 | **`gitlab-mr-commit`**       | OK      | Partially Supported [2] | Partially Supported [2] | Partially Supported [2] |
 | **`gerrit-change-review`**   | OK      | OK? [3]        | OK? [3]                 | Partially Supported? [2][3] |
 | **`bitbucket-code-report`**  | NO [4]  | NO [4]         | NO [4]                  | OK |
+| **`gitea-pr-review`**        | OK      | OK? [3]        | OK? [3]                 | OK |
```
Comment (haya14busa) [inline README.md:935]: It should be OK, OK, Partially Supported [2], Partially Supported [2]

**Review by haya14busa — CHANGES_REQUESTED** [review]
LGTM except the comment on Readme.

**Comment by haya14busa** [comment]
@lafriks I don't have any access to gitia. Did you test the code actually works with gitia? 
Can we potentially have some integration tests?

**Comment by lafriks** [comment]
I'm also a maintainer of Gitea, so I did a testing on my computer with the dev version. Are there any examples with integration tests? Is it possible to start a docker based service in GitHub actions?

**Comment by lafriks** [comment]
If you want to test you can also do that on https://try.gitea.io/ but I would not use it for integration tests

File: README.md (hunk 2 above)
Comment (lafriks) [reply README.md]: @haya14busa fixed

**Comment by haya14busa** [comment]
For GitLab, we have an integration test. We created a synced repo and use GitLab CI. https://github.com/reviewdog/reviewdog/blob/master/.gitlab-ci.yml

**Review by haya14busa — APPROVED** [review]

**Comment by haya14busa** [comment]
Thank you for your work！👍

**Comment by lafriks** [comment]
I see if I can do about implementing tests, at minimum unit tests

**Pull request merged**